Realm/RLMManagedDictionary.mm

```diff
@@ -102,6 +105,11 @@ - (RLMManagedDictionary *)initWithBackingCollection:(realm::object_store::Dictio
                                            property:(__unsafe_unretained RLMProperty *const)property {
     if (property.type == RLMPropertyTypeObject)
         self = [self initWithObjectClassName:property.objectClassName keyType:property.dictionaryKeyType];
+    else if (property.type == RLMPropertyTypeAny)
+        // Because the property is type mixed and we don't know if it will contain a dictionary when the schema
+        // is created, we set RLMPropertyTypeString by default.
+        // If another type is used for the dictionary key in a mixed dictionary context, this will thrown by core.
+        self = [self initWithObjectType:property.type optional:property.optional keyType:RLMPropertyTypeString];
```

Comment:
When we create the schema we don't know what it is going to contain, so when we create the dictionary the property type will be of type Any, and property key type will be set to nil. There are two options here, or set the key type to string even though we don't know what the user's keys types are and let core throw any errors otherwise, or skip validation for key type if the property type is Any.

Reply:
We are allowing users to assign NSArrays and NSDictionary so they can initialise mixed values easier, which could have keys of different type.